SYCL/ESIMD/thread_id_test.cpp

```diff
@@ -0,0 +1,97 @@
+// REQUIRES: gpu
+// UNSUPPORTED: cuda || hip
+// RUN: %clangxx -fsycl %s -o %t.out
+// RUN: %GPU_RUN_PLACEHOLDER %t.out
+// XFAIL: gpu
+//==- thread_id_test.cpp - Test to verify thread id functionlity-==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+// This is basic test to validate thread id functions.
+// TODO: Enable the test once the GPU RT supporting the functionality reaches
+// the CI
+
+#include <cmath>
+#include <iostream>
+#include <sycl/ext/intel/esimd.hpp>
+#include <sycl/ext/intel/esimd/simd.hpp>
+#include <sycl/sycl.hpp>
+#include <vector>
+
+int ErrCnt = 0;
+template <typename DataT>
+using shared_allocator = sycl::usm_allocator<DataT, sycl::usm::alloc::shared>;
+template <typename DataT>
+using shared_vector = std::vector<DataT, shared_allocator<DataT>>;
+
+int test_thread_id() {
+  sycl::queue queue;
+  shared_allocator<int32_t> allocator(queue);
+  constexpr int32_t SIZE = 32;
+
+  shared_vector<int32_t> vector_output_thread_id(SIZE, -1, allocator);
+  shared_vector<int32_t> vector_output_subdevice_id(SIZE, -1, allocator);
+
+  auto GlobalRange = sycl::range<1>(SIZE);
+  sycl::range<1> LocalRange{1};
+  sycl::nd_range<1> Range(GlobalRange, LocalRange);
+
+  {
+    auto e = queue.submit([&](sycl::handler &cgh) {

```

Comment:
minor comment: e is unused
```suggestion
    queue.submit([&](sycl::handler &cgh) {
```